src/packages/v4/hooks/useJBUpcomingRuleset.ts

```diff
@@ -0,0 +1,47 @@
+import { DecayRate, JBRulesetMetadata, RedemptionRate, ReservedRate, RulesetWeight } from 'juice-sdk-core';
+import { useJBContractContext, useReadJbControllerUpcomingRulesetOf } from 'juice-sdk-react';
+import { Ruleset } from '../models/ruleset';
+
+
+// @todo: add to SDK
+export function useJBUpcomingRuleset(): { 
+  ruleset: Ruleset | undefined, 
+  rulesetMetadata: JBRulesetMetadata | undefined, 
+  isLoading: boolean
+ } {
+  const { contracts, projectId } = useJBContractContext()
+  const { data, isLoading } = useReadJbControllerUpcomingRulesetOf({
+    address: contracts.controller?.data ?? undefined,
+    args: [projectId]
+  })
```

Comment:
no biggie but we can use the `select` function on this hook to make the below data transformations a lot easier. DW though, we'l move these to sdk eventually

Reply:
Ok got it 💯